### PR TITLE
Use keccak256 in tests and replace the SHA3 instruction in assembly

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/libsolidity/SolidityOptimizer.cpp [hunk 1]
```diff
@@ -1296,7 +1296,7 @@ BOOST_AUTO_TEST_CASE(constant_optimization_early_exit)
 			// Store and hash
 			assembly {
 				mstore(32, x)
-				ret := sha3(0, 40)
+				ret := keccak256(0, 40)
```
Comment (chriseth) [inline test/libsolidity/SolidityOptimizer.cpp:1299]: Please also add a test that shows that `sha3` is still available in inline assembly.

**Comment by axic** [comment]
Need to add changelog / documentation updates.

File: test/libsolidity/SolidityOptimizer.cpp (hunk 1 above)
Comment (axic) [reply test/libsolidity/SolidityOptimizer.cpp:1299]: Added.

**Comment by axic** [comment]
@chriseth this should be ready

**Review by chriseth — APPROVED** [review]

**Pull request merged**